presto-main/src/main/java/com/facebook/presto/execution/scheduler/nodeSelection/NodeSelectionStats.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.execution.scheduler.nodeSelection;
+
+import org.weakref.jmx.Managed;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+import java.util.concurrent.atomic.AtomicLong;
+
+@ThreadSafe
+public class NodeSelectionStats
+{
+    private final AtomicLong primaryHit = new AtomicLong();
```

Comment:
We shall give better names to these stats: (I'm not an expert on naming; maybe @shixuan-fan can give suggestions)
`primaryPreferredNodeSelectedCount`
`secondaryPreferredNodeSelectedCount`
`nonPreferredNodeSelectedCount`

Reply:
@highker I'm even less of an expert on naming than you :D Your suggestions look good to me.

Reply:
Do we want to use names like
primaryCandidateNodeSelectedCount
secondaryCandidateNodeSelectedCount
randomCandidateNodeSelectedCount